src/Client/HedgedConnections.cpp

```diff
@@ -195,6 +195,12 @@ void HedgedConnections::sendQuery(
             modified_settings.parallel_replica_offset = fd_to_replica_location[replica.packet_receiver->getFileDescriptor()].offset;
         }
 
+        /// FIXME: Remove once we will make `allow_experimental_analyzer` obsolete setting.
+        /// Make the analyzer being set, so it will be effectively applied on the remote server.
+        /// In other words, the initiator always controls whether the analyzer enabled or not for
+        /// all servers involved in the distributed query processing.
+        modified_settings.set("allow_experimental_analyzer", static_cast<bool>(modified_settings.allow_experimental_analyzer));
```

Comment:
We should not change the setting if `allow_experimental_analyzer == 0` and the server version is old enough (when the setting itself does not exist). 

Well, looks like it's anything older than 22.3, so maybe not very important
https://github.com/ClickHouse/ClickHouse/pull/42649